galite-core/src/main/kotlin/org/kopi/galite/preview/VPreviewWindow.kt

```diff
@@ -18,6 +18,378 @@
 
 package org.kopi.galite.preview
 
-class VPreviewWindow {
+import java.awt.event.KeyEvent
+import java.io.File
+import java.util.Locale
 
+import javax.swing.event.EventListenerList
+
+import org.kopi.galite.base.Utils
+import org.kopi.galite.l10n.LocalizationManager
+import org.kopi.galite.util.PrintJob
+import org.kopi.galite.visual.ApplicationContext
+import org.kopi.galite.visual.Constants
+import org.kopi.galite.visual.UIFactory
+import org.kopi.galite.visual.UWindow
+import org.kopi.galite.visual.VActor
+import org.kopi.galite.visual.VWindow
+import org.kopi.galite.visual.VlibProperties
+import org.kopi.galite.visual.WindowBuilder
+import org.kopi.galite.visual.WindowController
+
+/**
+ * A special window that display an html help
+ */
+class VPreviewWindow : VWindow() {
+  companion object {
+    private const val DEF_ZOOM_RATIO = 1.30f
+    private const val PREVIEW_LOCALIZATION_RESOURCE = "org/kopi/galite/Preview"
+
+    // the following commands *MUST* be in the same order than
+    // in 'actors' field set in the contructor of the current class.
+    const val CMD_QUIT = 0
+    const val CMD_FIRST = 1
+    const val CMD_LEFT = 2
+    const val CMD_RIGHT = 3
+    const val CMD_LAST = 4
+    protected const val CMD_ZOOM_FIT = 5
+    protected const val CMD_ZOOM_FIT_W = 6
+    protected const val CMD_ZOOM_FIT_H = 7
+    protected const val CMD_ZOOM_PLUS = 8
+    protected const val CMD_ZOOM_MINUS = 9
+
+    init {
+      WindowController.windowController.registerWindowBuilder(Constants.MDL_PREVIEW, object : WindowBuilder {
+        override fun createWindow(model: VWindow): UWindow {
+          return UIFactory.uiFactory.createView(model) as UWindow
+        }
+      })
+    }
+  }
+
+  override fun getType(): Int = Constants.MDL_PREVIEW
+
+  /**
+   * The user want to show an help
+   */
+  fun preview(printJob: PrintJob, command: String) {
+    val tempFile: File = Utils.getTempFile("PREVIEW", "JPG")
+    this.printJob = printJob
+    numberOfPages = printJob.numberOfPages
+    printFile = printJob.dataFile
```

Comment:
In printJob it's var datafile , so we need to change this one or change in printJob